src/control/cmd/dmg/storage_tables.go

```diff
@@ -0,0 +1,241 @@
+//
+// (C) Copyright 2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strings"
+
+	bytesize "github.com/inhies/go-bytesize"
+	"github.com/pkg/errors"
+
+	"github.com/daos-stack/daos/src/control/common/proto"
+	"github.com/daos-stack/daos/src/control/lib/hostlist"
+	"github.com/daos-stack/daos/src/control/server/storage"
+)
+
+func storageSummaryTable(hostsetTitle, scmTitle, nvmeTitle string, groups hostlist.HostGroups) (string, error) {
+	formatter := NewTableFormatter([]string{hostsetTitle, scmTitle, nvmeTitle})
+	var table []TableRow
+
+	for _, result := range groups.Keys() {
```

Comment:
yes to always ensure consistent ordering, @mjmac preferred this approach with maps rather than "slicing and dicing" in another area of code so thought I would follow suit.